test/benchmarks/material/slide-toggle/slide-toggle.perf-spec.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {$, browser} from 'protractor';
+import {runBenchmark} from '@angular/dev-infra-private/benchmark/driver-utilities';
+
+describe('slide toggle performance benchmarks', () => {
+  beforeAll(() => {
+    browser.rootEl = '#root';
+  });
+
+  it('renders a slide toggle', async() => {
+    await runBenchmark({
+      id: 'slide-toggle-render',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      prepare: async () => await $('#hide').click(),
+      work: async () => await $('#show').click(),
+    });
+  });
+
+  it('clicks a slide toggle', async() => {
+    await runBenchmark({
+      id: 'slide-toggle-click',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async () => await $('#show').click(),
+      work: async () => await $('.mat-slide-toggle-bar').click(),
```

Comment:
Could we just click on the `.mat-slide-toggle` element? That would match the most with an actual user-interaction. The click should be captured by the label element then (since protractor performs a real mouse click at center).

Reply:
clicking the `.mat-slide-toggle` element doesn't trigger the toggle

Reply:
That's surprising, but let's not bother with it then. I'd expect Protractor to just click centered in the toggle as a real user would do.

Reply:
Okay. Looks like selenium click is actually emitting a fake event. That explains why it doesn't simulate a real click.